__init__.py

```diff
@@ -2,15 +2,13 @@
 import ranger.api
 import ranger.core.fm
 import ranger.ext.signals
-from subprocess import Popen, PIPE
+from subprocess import Popen, PIPE, run
 
 hook_init_prev = ranger.api.hook_init
 
 def hook_init(fm):
     def zoxide_add(signal):
-        path = signal.new.path
-        process = Popen(["zoxide", "add", path])
-        process.wait()
+        Popen(["zoxide", "add", signal.new.path])
```

Comment:
Do we not want the `process.wait()` here?

Reply:
I wouldn't - we just want to add the path to zoxide and don't need a result. So doing it non-blocking doesn't hurt us, but is potentially faster 🙂

Reply:
What if it errors?

Reply:
Good question. In my eyes, there's nothing we can do if it errors (besides maybe putting out some message). Since the error probability should be quite low (it's been working for me for some months like this), the user should notice the error when `cd`ing anyway, and no sensible way to deal with the error, I would argue in favor of speed.
What's your opinion?

Reply:
Makes sense. 

Looks like regardless of the wait ranger provides the user with an error if the Popen returns a non-zero exit code. If it raises an uncaught OSError, ranger apparently crashes. Maybe we should catch that?

After reading some of the docs on [Popen](https://docs.python.org/3/library/subprocess.html), I wonder if we should use a `with` statement to ensure file descriptors are properly cleaned up.

Reply:
> Looks like regardless of the wait ranger provides the user with an error if the Popen returns a non-zero exit code. If it raises an uncaught OSError, ranger apparently crashes. Maybe we should catch that?

That does sound sensible to me.

> After reading some of the docs on Popen, I wonder if we should use a with statement to ensure file descriptors are properly cleaned up.

Sorry, I don't think I understand what you mean - `Popen` doesn't do anything with files in our case, does it? Which file descriptors are there to clean up?

Reply:
By file descriptors I mean stdout, stderr, and stdin. Perhaps the GC will clean them up, but I saw this in the docs:

> Popen objects are supported as context managers via the with statement: on exit, standard file descriptors are closed, and the process is waited for.

Reply:
Ah, I see. Not sure if we really need that, but if you want, I'll add them 🙂

Reply:
So I tried this out, but I don't think we need it here. The code would look like this:
```
with Popen(["zoxide", "add", signal.new.path]):
    pass
```

To me, that strongly suggests that it's useful if you read from `stdout`, for example, but here, we only want to call that command and immediately forget about it again.